PurchasesCoreSwift/SubscriberAttributes/SubscriberAttributesManager.swift

```diff
@@ -0,0 +1,261 @@
+    @objc public func setFBAnonymousID(_ maybeFBAnonymousID: String?, appUserID: String) {
+        Logger.debug(String(format: Strings.attribution.method_called, "setFBAnonymousID"))
+        setAttributionID(networkID: maybeFBAnonymousID,
+                         networkKey: SpecialSubscriberAttributes.fBAnonID,
+                         appUserID: appUserID)
+    }
+
+    @objc public func setMparticleID(_ maybeMparticleID: String?, appUserID: String) {
+        Logger.debug(String(format: Strings.attribution.method_called, "setMparticleID"))
+        setAttributionID(networkID: maybeMparticleID,
+                         networkKey: SpecialSubscriberAttributes.mpParticleID,
+                         appUserID: appUserID)
+    }
+
+    @objc public func setOnesignalID(_ maybeOnesignalID: String?, appUserID: String) {
+        Logger.debug(String(format: Strings.attribution.method_called, "setOnesignalID"))
+        setAttributionID(networkID: maybeOnesignalID,
+                         networkKey: SpecialSubscriberAttributes.oneSignalID,
+                         appUserID: appUserID)
+    }
+
+    @objc public func setMediaSource(_ maybeMediaSource: String?, appUserID: String) {
+        Logger.debug(String(format: Strings.attribution.method_called, "setMediaSource"))
+        setAttribute(key: SpecialSubscriberAttributes.mediaSource, value: maybeMediaSource, appUserID: appUserID)
+    }
+
+    @objc public func setCampaign(_ maybeCampaign: String?, appUserID: String) {
+        Logger.debug(String(format: Strings.attribution.method_called, "setCampaign"))
+        setAttribute(key: SpecialSubscriberAttributes.campaign, value: maybeCampaign, appUserID: appUserID)
+    }
+
+    @objc public func setAdGroup(_ maybeAdGroup: String?, appUserID: String) {
+        Logger.debug(String(format: Strings.attribution.method_called, "setAdGroup"))
+        setAttribute(key: SpecialSubscriberAttributes.adGroup, value: maybeAdGroup, appUserID: appUserID)
+    }
+
+    @objc public func setAd(_ maybeAd: String?, appUserID: String) {
+        Logger.debug(String(format: Strings.attribution.method_called, "setAd"))
+        setAttribute(key: SpecialSubscriberAttributes.ad, value: maybeAd, appUserID: appUserID)
+    }
+
+    @objc public func setKeyword(_ maybeKeyword: String?, appUserID: String) {
+        Logger.debug(String(format: Strings.attribution.method_called, "setKeyword"))
+        setAttribute(key: SpecialSubscriberAttributes.keyword, value: maybeKeyword, appUserID: appUserID)
+    }
+
+    @objc public func setCreative(_ maybeCreative: String?, appUserID: String) {
+        Logger.debug(String(format: Strings.attribution.method_called, "setCreative"))
+        setAttribute(key: SpecialSubscriberAttributes.creative, value: maybeCreative, appUserID: appUserID)
+    }
+
+    @objc public func collectDeviceIdentifiers(forAppUserID appUserID: String) {
+        Logger.debug("collectDeviceIdentifiers called")
+        // TODO safe to remove log below?
```

Comment:
this log seemed redundant to me since the calls to setAttribute below will also log...

Reply:
💯  💯